Tests/PSGetUpdateModuleManifest.Tests.ps1

```diff
@@ -896,4 +899,68 @@ Describe PowerShell.PSGet.UpdateModuleManifest -Tags 'BVT','InnerLoop' {
     } `
     -Skip:$($PSVersionTable.PSVersion -lt '5.1.0')
-}
\ No newline at end of file
+
+    # Purpose: Validate Update-ModuleManifest cmdlet throws warnings when passed a prerelease string with invalid characters
+    #
+    # Action:
+    #      Update-ModuleManifest -Path [Path] -PreRelease '-alpha+001'
+    #
+    # Expected Result: Update-ModuleManifest should throw InvalidCharactersInPreReleaseString errorid.
+    #
+    It UpdateModuleManifestWithInvalidPreReleaseString {
+        $PreRelease = "-alpha+001" 
+        $Version = "3.2.1"
+
+        $expectedErrorMessage = "The PreRelease string contains invalid characters. Please ensure that only characters 'a-zA-Z0-9' and possibly hyphen ('-') at the beginning are in the PreRelease string."
+        $expectedFullyQualifiedErrorId = "InvalidCharactersInPreReleaseString,Validate-PreReleaseString"
+
+        $ScriptBlock = {
+            New-ModuleManifest -path $script:testManifestPath
+            Update-ModuleManifest -Path $script:testManifestPath -PreRelease $PreRelease -ModuleVersion $Version -Confirm:$false
+        }
+
+        $ScriptBlock | Should -Throw $expectedErrorMessage -ErrorId $expectedFullyQualifiedErrorId
+    } 
+
+    # Purpose: Validate Update-ModuleManifest cmdlet throws warnings when passed a prerelease string when the version is too short.
+    #
+    # Action:
+    #      Update-ModuleManifest -Path [Path] -ModuleVersion '3.2' -PreRelease '-alpha001'
+    #
+    # Expected Result: Update-ModuleManifest should throw InsufficientVersionPartsForPreReleaseString errorid.
+    #
+    It UpdateModuleManifestWithPreReleaseStringAndShortModuleVersion {
+        $PreRelease = "-alpha001" 
+        $Version = "3.2"
+
+        $expectedErrorMessage = "Version has less than 3 parts and a PreRelease string is specified. Version must have at least 3 parts for a PreRelease string to be used."
+        $expectedFullyQualifiedErrorId = "InsufficientVersionPartsForPreReleaseStringUsage,Validate-PreReleaseString"
+
+        $ScriptBlock = {
+            New-ModuleManifest -path $script:testManifestPath
+            Update-ModuleManifest -Path $script:testManifestPath -PreRelease $PreRelease -ModuleVersion $Version -Confirm:$false
+        }
+
+        $ScriptBlock | Should -Throw $expectedErrorMessage -ErrorId $expectedFullyQualifiedErrorId
+    } 
+    
+    # Purpose: Validate Update-ModuleManifest cmdlet in PowerShell version greater than 5.0 when passed valid PreRelease string and module version.
+    #
+    # Action:
+    #      Update-ModuleManifest -Path [Path] -ModuleVersion '3.2.1' -PreRelease '-alpha001'
+    #
+    # Expected Result: Update-ModuleManifest should update the PreRelease and Version fields in module manifest file.
+    #
+    It UpdateModuleManifestWithValidPreReleaseAndModuleVersion {
+        $PreRelease = "-alpha001" 
```

Comment:
is it possible to add error testcases with patterns like -alpha-beta.01, -alpha.1, -gamma001, -error.0.0.01